### DISCUSSION TITLE
Data storage

### DISCUSSION THREAD

**Discussion by zibonbadi** [zibonbadi/liquidms#11] [General]
This issue is meant for discussion on how liquidMS will store it's information on available servers.

According to the official master server's API, this most likely requires a connection between liquidMS and a relational database.

**Comment by zibonbadi** [comment]
In order to decouple the project from any particular database implementation, I recommend using the PHP ODBC extension:

### Pros

- Allows the use of standardized interfaces and SQL queries to manage the database, guaranteeing interoperability.
- ODBC's driver model allows databases to be hosted arbitrarily, be it as a file, a separate program or remote
- Separation of concerns regarding the database

### Cons

- More involved deployment of the database (e.g. ODBC drivers required)
- Possible performance loss compared to vendor-specific implementations
- Possibly less advanced query behavior compared to vendor-specific implementations.
- Separate administration effort for the database

**Comment by thewane** [comment]
I agree on using ODBC, the pros simply outweigh the cons. The scope of this project being a decentralized network of master servers further helps this, along with how distributable a standardized API is (outside of drivers).

**Comment by zibonbadi** [comment]
# Announcement: Migration to liquidMS discussions

For the record, this topic has previously been kept as an issue. It has now (2021-11-26) been converted to a discussion.
